x-pack/plugins/monitoring/server/routes/api/v1/setup/cluster_setup_status.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import Joi from 'joi';
+import { verifyMonitoringAuth } from '../../../../lib/elasticsearch/verify_monitoring_auth';
+import { handleError } from '../../../../lib/errors';
+import { getCollectionStatus } from '../../../../lib/setup/collection';
+import { getIndexPatterns } from '../../../../lib/cluster/get_index_patterns';
+
+export function clusterSetupStatusRoute(server) {
+  /*
+   * Monitoring Home
+   * Route Init (for checking license and compatibility for multi-cluster monitoring
+   */
+  server.route({
+    method: 'POST',
+    path: '/api/monitoring/v1/setup/collection/{clusterUuid}',
```

Comment:
Couldn't you combine this with the `clustersSetupStatusRoute` route, by making `{clusterUuid}` optional? Everything else seems to be the same between the two route implementations.

Reply:
I suppose. I was following the same paradigm that the [clusters](https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/server/routes/api/v1/cluster/clusters.js) and [cluster](https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/server/routes/api/v1/cluster/cluster.js) routes do. I don't feel strongly either way

Reply:
Ah okay. I'd prefer to share code but I'm +1 to keeping things consistent in this PR. Perhaps we could do some refactoring in a follow up PR.

Reply:
@chrisronline WDYT about "collapsing" these very-similar routes into the same one in a follow up PR? If you agree I can file a cleanup/refactoring issue for it.

Reply:
I added a note here: https://github.com/elastic/stack-monitoring/issues/28. Feel free to file a ticket too if you want, but I think it's being tracked either way